autopilot/contractor/contractor.go

```diff
@@ -1199,7 +1199,7 @@ func performV2ContractMigration(ctx *mCtx, bus Database, cr contractReviser, cs
 		if err := bus.ArchiveContracts(ctx, map[types.FileContractID]string{
 			id: "migrated to v2",
 		}); err != nil {
-			logger.Errorf("failed to archive migrated contract")
+			logger.Errorw("failed to archive migrated contract", zap.Error(err))
```

Comment:
This is better but we're still inconsistent with `zap.Error`. I think this migration code could use a little more debug log lines as well. E.g. debug log the number of v1 contracts remaining. If the host already has a v2 contract and we archive the other v1 contract, debug log both fcids and archive this contract under a different reason.


In my refactor I made `toArchive := make(map[types.FileContractID]string)` which allows you to pass it to `bus.ArchiveContracts` in one go and archive them all together. I would also make `hostsWithV2Contract := make(map[types.PublicKey]types.FileContractID)` so we can cleanly log the v2 contract.

Reply:
I don't see the value of a different reason tbh. If anything unexpected happens we get an error now which imo is good enough.